packages/shared/social/avatarTracker.ts

```diff
@@ -1,17 +1,32 @@
 import { AvatarMessageType } from 'shared/comms/interface/types'
 import { getUser } from 'shared/comms/peers'
 import { avatarMessageObservable } from 'shared/comms/peers'
-import { allScenesEvent } from 'shared/world/parcelSceneManager'
+import { allScenesEvent, getSceneWorkerBySceneID } from 'shared/world/parcelSceneManager'
+import { Observable } from 'decentraland-ecs'
+import { AvatarRendererMessage, AvatarRendererMessageType } from 'shared/types'
 
-const avatarConnected = 'playerConnected'
-const avatarDisconnected = 'playerDisconnected'
-
-const visibleAvatars: Map<string, string> = new Map<string, string>()
+export const avatarRendererMessageObservable = new Observable<AvatarRendererMessage>()
 
 export function getVisibleAvatarsUserId(): string[] {
   return Array.from(visibleAvatars.values())
 }
 
+export function getInSceneAvatarsUserId(sceneId: string): string[] {
+  return Array.from(rendererAvatars.entries())
+    .filter(([userId, avatarData]) => avatarData.sceneId === sceneId)
+    .map(([userId, avatarData]) => userId)
+}
```

Comment:
The functional approach is heavy for the garbage collector and should be avoided when performance is critical.. 

```suggestion
export function* getInSceneAvatarsUserId(sceneId: string): Iterable<string> {
  for (const [userId, avatarData] of rendererAvatars) {
    if (avatarData.sceneId === sceneId) yield userId
  }
}
```

Then can be used with a `for` (optimal) or `Array.from(getInSceneAvatarsUserId())`